bin/webpack-dev-server.js

```diff
@@ -418,6 +407,27 @@ function startDevServer(config, options) {
 
       status(uri, options, log, argv.color);
     });
+  } else {
+    // only run port finder if no port as been specified
+    findPort(server, DEFAULT_PORT, defaultPortRetry, (err, port) => {
+      if (err) {
+        throw err;
+      }
+      options.port = port;
+      server.listen(options.port, options.host, (err) => {
```

Comment:
Maybe we can move `server.listen` to separate function, looks it is duplicate code

Reply:
Indeed, I'll look into that over the weekend. Making sure all the variables are passed in for the subtle differences in each server.listen call would be a bit tedious.

Reply:
Nvm... the server.listen uses for sockets and ports are too different. I just extract the 2 identical server.listens for port uses in to a local function to keep variable references. Not sure if that's aligned with your coding style standards though. Please comment if that also needs to be changed.

Reply:
@u9520107 hm, let's leave this as is, and refactor this in future